include/deal.II/lac/trilinos_vector.h

```diff
@@ -988,7 +988,7 @@ namespace internal
                                 TrilinosWrappers::MPI::Vector &v,
                                 bool fast)
       {
-        v.reinit(matrix.range_partitioner(), fast);
+        v.reinit(IndexSet(matrix.range_partitioner()), MPI_COMM_WORLD, fast);
```

Comment:
Yes, using MPI_COMM_WORLD is not acceptable. We need to get the communicator from somewhere else. You could also use `TrilinosWrappers::VectorBase::trilinos_vector()`, which gives you a `Epetra_FEVector` that likely has a function to give you the MPI communicator.